include/circt/Dialect/RTL/RTLStatements.td

```diff
@@ -48,6 +48,13 @@ def WireOp : RTLOp<"wire", [DeclareOpInterfaceMethods<OpAsmOpInterface>]> {
 
   let assemblyFormat = "custom<ImplicitSSAName>(attr-dict) `:` type($result)";
   let hasCanonicalizer = true;
+
+  let extraClassDeclaration = [{
+    Type getElementType() {
```

Comment:
nice.  I bet this could be used in a bunch of places to simplify code.  It would be worth looking for those places if you haven't already.